src/test/java/seedu/address/ui/testutil/GuiTestAssert.java

```diff
@@ -34,8 +42,59 @@ public static void assertCardDisplaysPerson(Person expectedPerson, PersonCardHan
         assertEquals(expectedPerson.getPhone().value, actualCard.getPhone());
         assertEquals(expectedPerson.getEmail().value, actualCard.getEmail());
         assertEquals(expectedPerson.getAddress().value, actualCard.getAddress());
-        assertEquals(expectedPerson.getTags().stream().map(tag -> tag.tagName).collect(Collectors.toList()),
-                actualCard.getTags());
+
+        assertTagsEquals(expectedPerson, actualCard);
+    }
+
+    /**
+     * Returns the color style for {@code tagName}'s label. The tag's color is determined by looking up the color
+     * in {@code PersonCard#TAG_COLOR_STYLES}, using an index generated by the hash code of the tag's content.
+     *
+     * @see PersonCard#getTagColorStyleFor(String)
+     */
+    private static String getTagColorStyleFor(String tagName) {
+        switch (tagName) {
+        case "classmates":
+            return "brown";
+
+        case "owesMoney":
+            return "green";
+
+        case "colleagues":
+            return "purple";
+
+        case "neighbours":
+            return "yellow";
+
+        case "family":
+            return "lightgreen";
+
+        case "friend":
+            return "grey";
+
+        case "friends":
+            return "pink";
+
+        case "husband":
+            return "red";
+
+        default:
+            fail(tagName + " does not have a color assigned.");
+            return "";
+        }
+    }
+
+    /**
+     * Asserts that the tags in {@code actualCard} matches all the tags in {@code expectedPerson} with the correct
+     * color.
+     */
+    private static void assertTagsEquals(Person expectedPerson, PersonCardHandle actualCard) {
+        List<String> expectedTags = expectedPerson.getTags().stream()
+                .map(tag -> tag.tagName).collect(Collectors.toList());
+        assertEquals(expectedTags, actualCard.getTags());
+        expectedTags.forEach(tag ->
+            assertEquals(Arrays.asList(LABEL_DEFAULT_STYLE, getTagColorStyleFor(tag)),
+                    actualCard.getTagStyleClasses(tag)));
```

Comment:
here the line wrap is a bit bad 
Perhaps something like this would be better.
<pre>
expectedTags.forEach(tag -> assertEquals(
        Arrays.asList(LABEL_DEFAULT_STYLE, getTagColorStyleFor(tag)), actualCard.getTagStyleClasses(tag)));
</pre>